scripts/azure_deployment/arm_aci.py

```diff
@@ -88,73 +89,130 @@ def make_aci_deployment(parser: ArgumentParser) -> Deployment:
                                 "ports": [
                                     {"protocol": "TCP", "port": 8000},
                                     {"protocol": "TCP", "port": 22},
                                 ],
-                                "type": "Public",
+                                "environmentVariables": [],
+                                "resources": {"requests": {"memoryInGB": 16, "cpu": 4}},
+                            },
+                        }
+                    ],
+                    "ipAddress": {
+                        "ports": [
+                            {"protocol": "TCP", "port": 8000},
+                            {"protocol": "TCP", "port": 22},
+                        ],
+                        "type": "Public",
+                    },
+                },
+            }
+            for i in range(args.count)
+        ],
+    }
+
+    if args.aci_deploy_external_executer:
+        template["resources"].append(
+            {
+                **common_resource_attributes,
+                "name": f"{args.deployment_name}-business-logic",
+                "properties": {
+                    **common_resource_properties,
+                    "containers": [
+                        {
+                            "name": f"{args.deployment_name}-attestation-container",
+                            "properties": {
+                                "image": f"attestationcontainerregistry.azurecr.io/attestation-container:{args.deployment_name}",
+                                "command": [
+                                    "/bin/sh",
+                                    "-c",
+                                    " && ".join(
+                                        [
+                                            *STARTUP_COMMANDS[args.aci_type](
+                                                args,
+                                                None,
+                                            ),
+                                            "app",
+                                        ]
+                                    ),
+                                ],
+                                "ports": [
+                                    {"protocol": "TCP", "port": 22},
+                                    {"protocol": "TCP", "port": 50051},
+                                ],
+                                "environmentVariables": [],
+                                "resources": {"requests": {"memoryInGB": 16, "cpu": 4}},
                             },
-                            "osType": "Linux",
-                        },
-                    }
-                    for i in range(args.count)
-                ],
-            },
+                        }
+                    ],
+                    "ipAddress": {
+                        "ports": [
+                            {"protocol": "TCP", "port": 22},
+                            {"protocol": "TCP", "port": 50051},
```

Comment:
Opening attestation-container gRPC's port is for running e2e test from CI agent. But considering that we should not expose attestation-container in the actual application, I will close this port and run e2e test inside the attestation-container using ssh + port 22 (Port 22 shouldn't be exposed in the actual application as well, but I think it's more obvious and hopefully it prevents application developers from exposing attestation-container for outside of the container group by mistake).

Reply:
Added comment in 7e9d165.